.github/workflows/umd-unit-tests.yaml

```diff
@@ -0,0 +1,59 @@
+name: "[internal] UMD Unit tests"
+
+on:
+  workflow_call:
+    inputs:
+      arch:
+        required: true
+        type: string
+      runner-label:
+        required: true
+        type: string
+      timeout:
+        required: false
+        type: number
+        default: 15
+  workflow_dispatch:
+    inputs:
+      arch:
+        required: true
+        type: choice
```

Comment:
Argh I wonder if there's any way to reduce this copy pasta - not sure if we can use the YAML anchor thing @ttmchiou